### ISSUE TITLE
Block/Layer name

### ISSUE THREAD

**Issue by som3on3** [arklumpus/MuPDFCore#36]
```
using var ctx = new MuPDFContext();
using var document = new MuPDFDocument(ctx, "source.pdf");

var page = document.GetStructuredTextPage(0);
foreach (var block in page) {
    // how to get layer name??
}
```
How to get the layer name from block?

**Comment by arklumpus** [comment]
Hi! I think you are referring to optional content groups (sometimes also known as layers), right? I.e., bits of content that can be toggled on and off in the PDF viewer interface. If this is the case, these are not currently supported by this library, but I can work on it as soon as I have a bit of time.

The structured text blocks you get from the structured text page are completely unrelated to this. Also note that the `MuPDFStructuredTextPage` class implements `IDisposable`, therefore you should also make sure that it is `Dispose`d - e.g., with `using var page = document.GetStructuredTextPage(0);`

I will leave this issue open until I manage to look at implementing optional content groups!

**Comment by arklumpus** [comment]
Hi, optional content groups/layers are finally implemented in MuPDFCore v2.0.0! You can access them from the `OptionalContentGroupData` property of the `MuPDFDocument` class:

```
using var ctx = new MuPDFContext();
using var document = new MuPDFDocument(ctx, "source.pdf");

MuPDFOptionalContentGroup[] layers = document.OptionalContentGroupData.OptionalContentGroups;

foreach (MuPDFOptionalContentGroup layer in layers)
{
    Console.WriteLine(layer.Name);
}
```

I hope this helps! I'm closing this now, feel free to reopen it (or create a new issue) if you have any more questions!
